test/libdevcore/IpfsHash.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Unit tests for the ipfs hash computation routine.
+ */
+
+#include <libdevcore/IpfsHash.h>
+
+#include <test/Options.h>
+
+using namespace std;
+
+namespace dev
+{
+namespace test
+{
+
+BOOST_AUTO_TEST_SUITE(IpfsHash)
+
+BOOST_AUTO_TEST_CASE(test_small)
+{
+	BOOST_CHECK_EQUAL(ipfsHashBase58({}), "QmbFMke1KXqnYyBBWxB74N4c5SBnJMVAiMNRcGu6x1AwQH");
+	BOOST_CHECK_EQUAL(ipfsHashBase58("x"), "QmULKig5Fxrs2sC4qt9nNduucXfb92AFYQ6Hi3YRqDmrYC");
```

Comment:
I'm getting different values when running
```
echo "x" | ipfs add
added QmUNXr47Bja3aHUMfhXX5mMWTFJKuoUGETcA48vHG7dhag
```
Am I missing smth here?

Reply:
```
echo -n "x" | ipfs add
added QmULKig5Fxrs2sC4qt9nNduucXfb92AFYQ6Hi3YRqDmrYC 
```

Reply:
ok that works